### PR TITLE
api.raml adjustment for vm update

### PR DESCRIPTION
**Pull request by abdulrahmantkhalifa**
PUT node/<nodeid>/vm/<vmid>/volume/<volumesid

### CODE REVIEW

File: raml/api.raml [hunk 1]
```diff
@@ -670,6 +670,18 @@ types:
...
                 application/json:
                   type: VM
+        put:
+          displayName: UpdateVm
+          description: updates the VM 
+          body:
+            application/json:
+              type: VMBase
+          responses:
+            200:
```
Comment (FastGeert) [inline raml/api.raml]: I don't see the point of returning this information

File: raml/types.raml [hunk 2]
```diff
@@ -725,9 +719,9 @@ VMInfo:
   type: object
   description: Statistical information about a vm
   properties:
-    disk: VMDiskInfo[]
+    disks: VMDiskInfo[]
     net: VMNetInfo[]
-    cpu:
+    cpus:
```
Comment (FastGeert) [inline raml/types.raml]: cpus is not a collection, should be cpu

File: raml/types.raml [hunk 3]
```diff
@@ -255,19 +255,14 @@ VMListItem:
...
-      type: string
-    status:
-      enum: [running, halted, paused, halting, migrating]
-      description: Status of the VM
+    name: string  
     memory:
       description: Amount of memory in MiB
       type: integer
-    cpu:
+    cpus:
```
Comment (FastGeert) [inline raml/types.raml]: Should be singular, we use plural only for collections

File: raml/types.raml [hunk 4]
```diff
@@ -277,22 +272,21 @@ VM:
       description: List of virtual disks
       type: VDiskLink[]
 
+
+VM:
+  type: VMBase
+  properties:
+    id:
```
Comment (FastGeert) [inline raml/types.raml]: id should be in VMBase

File: raml/types.raml [hunk 5]
```diff
@@ -725,9 +719,9 @@ VMInfo:
   type: object
   description: Statistical information about a vm
   properties:
-    disk: VMDiskInfo[]
+    disks: VMDiskInfo[]
     net: VMNetInfo[]
```
Comment (FastGeert) [inline raml/types.raml]: net --> nics

VMNetInfo --> VMNicInfo

File: raml/types.raml [hunk 6]
```diff
@@ -300,6 +300,24 @@ VMCreate:
       description: Systemlevel cloudinit configuration
       type: object
 
+VMUpdate:
+  type: object
+  properties:
+    id: string
```
Comment (FastGeert) [inline raml/types.raml]: id not necessary here, will be in the url

Comment (abdulrahmantkhalifa) [inline raml/types.raml]: will it not change? i mean can't ovc need to change name ?

Comment (FastGeert) [inline raml/types.raml]: no changing id's is something we don't need

**Pull request merged**